Sources/GRPCNIOTransportHTTP2Posix/HTTP2ServerTransport+Posix.swift

```diff
@@ -186,6 +199,15 @@ extension HTTP2ServerTransport {
 }
 
 extension HTTP2ServerTransport.Posix {
+  /// Context for Posix TransportSpecific
+  public struct Context: ServerContext.TransportSpecific {
```

Comment:
How does this compose overall? Shouldn't this be something that really every transport can vend optionally so code that has to do authn/z doesn't have to become transport specific @glbrntt?

Reply:
The longer term move is to have mixin protocols (e.g. `HasPeerCertificate`) which contexts would conform to. Any common code would then use these rather than relying on a concrete transport specific type.

Reply:
I guess my meta question is why do we consider that peer certificate needs to be a mix-in protocol and we can't make it part of the `ServerContext` right away. It seems in line with `remotePeer: String` to have a `remotePeerCertificate: Certificate?`. Since it has to be optional anyways.

Reply:
It isn't really in line with it, you can't compute on `remotePeer` beyond using it as an identifier. `remotePeer` is also available to all transports, whereas the peer cert is only available here.

Adding it to the context brings in three more dependencies which would only be used by the NIO transport.

Reply:
Why only the NIO transport? Can't we provide a `Certificate` based peer identity on the `NIOTS` transport as well?

Reply:
NIO transport includes NIOTS. The NIOTS transport work hasn't been done yet (I'm not sure if the right APIs currently exist to do it.)